src/vsphere_cpi/spec/integration/set_vm_metadata_spec.rb

```diff
@@ -57,6 +113,30 @@
       expect {
         threads.map(&:join)
       }.to_not raise_error
+
+    end
+  end
+
+  context 'when attaching tags and custom fields to VM' do
+    subject(:cpi) { VSphereCloud::Cloud.new(cpi_options) }
+    it 'assigns relevant tags on VM' do
+      vm_cid = @cpi.create_vm(
+          'agent-007',
+          @stemcell_id,
+          vm_type,
+          network_spec,
+          [],
+          {}
+      )
+      @cpi.set_vm_metadata(vm_cid, metadata)
+
+      #expect current vm to have tags attached
+      mob_id = @cpi.vm_provider.find(vm_cid).mob.__mo_id__
+      expect(vm_cid).to_not be_nil
+      expect(verify_tags(mob_id, ['Linux', 'MAPBU', 'vcpi-team'])).to be(true)
+      expect(verify_tags(mob_id, ['Type', 'deployment'])).to be(false)
```

Comment:
In a subsequent PR after merging this, you can look into writing a custom matcher: https://relishapp.com/rspec/rspec-expectations/v/3-5/docs/custom-matchers/define-a-custom-matcher#with-multiple-args

It can look something like `expect(vm).to_have_tags(a,b,c)`